### DISCUSSION TITLE
Adding Mercurius Explain Graphiql Plugin

### DISCUSSION THREAD

**Discussion by bilalebi** [mirumee/ariadne#1087] [Ideas]
I was looking for a way to monitor/profile graphql query and found [Mercurius Explain Graphiql Plugin](https://github.com/nearform/mercurius-explain-graphiql-plugin) which looks promising.

![Mercurius Explain Graphiql Plugin UI](https://github.com/nearform/mercurius-explain-graphiql-plugin/raw/master/docs/profiler.png)

Will it be possible to make it available through Ariadne?

**Comment by rafalp** [comment]
I've looked at their repo and from what I see this plugin is not just GraphiQL plugin but it also requires server to expose additional data about the query execution through [`mercurius-explain`](https://github.com/nearform/mercurius-explain) server plugin.

This is out of Ariadne's scope, but our extension system should enable others to implement their plugins to do things like this.

**Reply by bilalebi** [reply]
Thank you for your prompt reply, I'll take the extension system road
